src/executor/graph_executor.cc

```diff
@@ -662,7 +671,7 @@ void GraphExecutor::Init(nnvm::Symbol symbol,
   // This function can be called by regular bind
   // operation flow as well.
 #if MXNET_USE_NGRAPH == 1
-  if (multi_context) {
+  if (!multi_context && !grad_sparse) {
```

Comment:
I'm confused about the logic change (the '!') for multi_context.  Why did this change?

Reply:
Looks like a typo on my part in the original. Not sure why we haven't hit a bug here.

Reply:
i believe this was missing originally , @mbrookhart  can clarify.